### ISSUE TITLE
indentation errors exist in docs

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4733]
|<img alt="magcius's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| magcius reported|
|-|-|
|Trac ID|trac#4733|
|Type|enhancement|
|Created|2010-11-08 21:58:14Z|
|Branch|https://github.com/twisted/twisted/tree/doc-indentation-4733|

http://twistedmatrix.com/documents/10.1.0/web/howto/web-in-60/http-auth.html

look at the method here, it took me quite a while to figure out it's a method because the most important definer of being a method, the indentation, is gone

Attachments:

* [4733-fix-method-indent.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/69b/69be86e55f2f7b1d0eb179c68a4bbcc206ba9572/8b5e747e52c1ef33fc2cb5a631656ab5e234485a.patch) (866 bytes) - added by ivank on 2010-11-08 22:03:01Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4733 4733
type__enhancement enhancement
reporter__magcius magcius
priority__normal normal
milestone__ 
branch__branches_doc_indentation_4733 branches/doc-indentation-4733
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1289253494000000 1289253494000000
changetime__1300082263000000 1300082263000000
version__None None
owner__ 
cc__ivank cc__magcius cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ivank set owner to @jyknight|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set owner to spiv|
|-|-|

This is not a bug in lore, it's a bug in the source document:
http://twistedmatrix.com/trac/browser/trunk/doc/web/howto/web-in-60/http-auth.xhtml#L82

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst set owner to @jyknight|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

Just a note for future readers of this ticket, it is not okay to file a ticket like this.

**Comment by twisted-trac** [comment]
|<img alt="magcius's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|magcius commented|
|-|-|

quick browse finds other pages like this:

http://twistedmatrix.com/documents/10.1.0/web/howto/web-in-60/asynchronous-deferred.html

a quick grep:

```
$ grep -lR '^def .*(self' . | grep -v 'svn'
./words/howto/im.xhtml
./core/development/policy/test-standard.xhtml
./core/development/policy/coding-standard.xhtml
./core/examples/pyuidemo.py
./core/howto/testing.xhtml
./core/howto/process.xhtml
./web/howto/web-in-60/asynchronous-deferred.xhtml
./web/howto/web-in-60/handling-posts.xhtml
./web/howto/web-in-60/interrupted.xhtml
./web/howto/web-in-60/logging-errors.xhtml
./web/howto/web-in-60/http-auth.xhtml
```

changing summary, but there is really no general "docs" component, so staying as web

I'll have a patch in a bit.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_8"></a>|@jesstess set owner to magcius|
|-|-|

Removing from review since this is waiting on a new patch.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

(In [30905]) Branching to 'doc-indentation-4733'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun commented|
|-|-|

(In [30906]) Apply 4733-fix-method-indent.patch 

refs [#4733](https://github.com/twisted/twisted/issues/4733)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun commented|
|-|-|

(In [30907]) Add a class statement to contain the example test method.


This makes it more clear what you should be doing, both by indenting the method like you might expect to encounter it in the wild, and showing you that tests are methods on classes.


refs [#4733](https://github.com/twisted/twisted/issues/4733)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun commented|
|-|-|

(In [30909]) Similar change to indentation of sample code

refs [#4733](https://github.com/twisted/twisted/issues/4733)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun commented|
|-|-|

(In [30910]) I mean, similar indentation fix.

refs [#4733](https://github.com/twisted/twisted/issues/4733)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun commented|
|-|-|

(In [30911]) Fix indentation on one last http example

refs [#4733](https://github.com/twisted/twisted/issues/4733)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_15"></a>|@exarkun commented|
|-|-|

(In [30912]) Apply that same sort of fix

refs [#4733](https://github.com/twisted/twisted/issues/4733)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_16"></a>|@exarkun commented|
|-|-|

(In [30913]) Same indentation fix again

refs [#4733](https://github.com/twisted/twisted/issues/4733)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_17"></a>|@exarkun commented|
|-|-|

(In [30914]) And indent the rest of the lines of this function, in their separate code blocks

refs [#4733](https://github.com/twisted/twisted/issues/4733)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_18"></a>|@exarkun removed owner|
|-|-|

The approach in the original patch was unfortunately not valid.  Lore does not allow this style of leading indentation (we actually used to have a lot of it, and the output is still wrong, so we got rid of it all).

I put some changes in the branch that maybe improve the situation.  It forces indentation by either adding a class on top of a naked method or sticking a "..." in front to force the indentation.

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_19"></a>|@lvh commented|
|-|-|

I'm merging this because it's an improvement for Lore, but the Sphinx generated output is somewhat unsatisfactory so I also created [#4942](https://github.com/twisted/twisted/issues/4942).

**Comment by twisted-trac** [comment]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)<a name="note_20"></a>|@lvh set status to `closed`|
|-|-|

(In [31006]) Better indentation for HTTP authentication documentation


Author: exarkun
Reviewer: lvh
Fixes: [#4733](https://github.com/twisted/twisted/issues/4733)

There was a function in the Twisted web in 60 seconds documentation on HTTP Authentication had a method that looked like a function due to indentation. This is now fixed. This doesn't work well for the Sphinx generated docs, though. Sphinx has a nicer way of fixing this problem, so there's a separate ticket, [#4942](https://github.com/twisted/twisted/issues/4942).
